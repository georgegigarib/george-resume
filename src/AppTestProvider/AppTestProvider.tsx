import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
