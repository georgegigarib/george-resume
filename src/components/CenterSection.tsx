import React from 'react';
import Box from "./Box";
import { useTranslation } from 'react-i18next';


const CenterSection = () => {
const { t } = useTranslation()


    const boxes = [
        { startRow: 28, spanRow: 28, startColumn: 12, spanColumn: 22, content: t('hello') },
        { startRow: 56, spanRow: 9, startColumn: 25, spanColumn: 9, content: "Box 8" },
        { startRow: 51, spanRow: 16, startColumn: 34, spanColumn: 22, content: "Box 9" },
        { startRow: 33, spanRow: 18, startColumn: 34, spanColumn: 57, content: "Box 10" },
        { startRow: 18, spanRow: 15, startColumn: 48, spanColumn: 16, content: "Box 11" },
    ];

    return (
    <div className="grid grid-cols-100 grid-rows-100 gap-4 md:grid-cols-50 sm:grid-cols-20 w-full">
            {boxes.map((box, index) => (
                <Box 
                    key={index}
                    startRow={box.startRow}
                    startColumn={box.startColumn}
                    spanRow={box.spanRow}
                    spanColumn={box.spanColumn}
                    content={box.content} 
                />
            ))}
        </div>
    );
}

export default CenterSection;
