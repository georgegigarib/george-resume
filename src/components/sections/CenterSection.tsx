import { useState, useEffect } from 'react'
import { Box, BoxProps } from '@/components/organisms/Box'
import { useTranslation } from 'react-i18next'
import ExampleComponent from '@/components/organisms/ExampleComponent'
import LatestSong from '@/components/organisms/LatestSong'
import { useIsMobile } from '@/hooks/useIsMobile'
import Presentation from '@/components/organisms/Presentation'
import Jokes from '@/components/organisms/Jokes'

const CenterSection = () => {
  const { t, i18n } = useTranslation()
  const isMobile = useIsMobile()
  const [activeBoxes, setActiveBoxes] = useState<BoxProps[]>([])

  useEffect(() => {
    const mobileBoxes = [
      { startRow: 8, spanRow: 7, startColumn: 4, spanColumn: 16, content: <Presentation /> },
      { startRow: 15, spanRow: 14, startColumn: 9, spanColumn: 11, content: <Jokes /> },
      { startRow: 12, spanRow: 8, startColumn: 20, spanColumn: 8, content: <ExampleComponent /> },
      { startRow: 8, spanRow: 4, startColumn: 20, spanColumn: 21, content: <LatestSong /> },
    ]

    const desktopBoxes = [
      { startRow: 26, spanRow: 30, startColumn: 19, spanColumn: 18, content: <ExampleComponent /> },
      { startRow: 56, spanRow: 9, startColumn: 27, spanColumn: 9, content: <ExampleComponent /> },
      { startRow: 50, spanRow: 10, startColumn: 37, spanColumn: 22, content: <Jokes /> },
      { startRow: 31, spanRow: 19, startColumn: 37, spanColumn: 51, content: <ExampleComponent /> },
      { startRow: 22, spanRow: 9, startColumn: 43, spanColumn: 22, content: <Presentation /> },
    ]

    setActiveBoxes(isMobile ? mobileBoxes : desktopBoxes)
  }, [isMobile, i18n.language, t])

  return (
    <div className="grid grid-cols-2 grid-rows-20 lg:grid-cols-100 lg:grid-rows-100 gap-4">
      {activeBoxes.map((box, index) => (
        <Box
          key={index}
          startRow={box.startRow}
          startColumn={box.startColumn}
          spanRow={box.spanRow}
          spanColumn={box.spanColumn}
          content={box.content}
        />
      ))}
    </div>
  )
}

export default CenterSection
