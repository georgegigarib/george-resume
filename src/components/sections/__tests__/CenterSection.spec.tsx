import { render } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'
import CenterSection from '@/components/sections/CenterSection'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'

const isMobileMock = vi.fn()

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: () => isMobileMock()
}))

describe('CenterSection component', () => {
  const renderComponent = () =>
    render(
      <AppTestProvider>
        <CenterSection />
      </AppTestProvider>
    )

  it('renders correctly on desktop', () => {
    isMobileMock.mockReturnValue(false)

    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})
