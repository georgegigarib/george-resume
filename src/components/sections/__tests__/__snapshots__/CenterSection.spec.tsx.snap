// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CenterSection component > renders correctly on desktop 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-2 grid-rows-20 lg:grid-cols-100 lg:grid-rows-100 gap-4"
  >
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 26 / 19 / span 30 / span 18;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="text-blue-900 rounded-xl flex justify-center flex-wrap items-center contain-size overflow-hidden h-full border-[2px] bg-app-lightModeBoxes-box2 dark:bg-app-darkModeBoxes-box2 border-app-darkModeBoxes-box2 dark:border-app-lightModeBoxes-box2"
              data-testid="example-container"
            >
              <div
                class="absolute inset-0 flex items-center justify-center"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              <img
                class="rounded-lg min-h-full min-w-full flex-shrink object-cover opacity-0"
                src="/src/assets/images/construction.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 56 / 27 / span 9 / span 9;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="text-blue-900 rounded-xl flex justify-center flex-wrap items-center contain-size overflow-hidden h-full border-[2px] bg-app-lightModeBoxes-box1 dark:bg-app-darkModeBoxes-box1 border-app-darkModeBoxes-box1 dark:border-app-lightModeBoxes-box1"
              data-testid="example-container"
            >
              <div
                class="absolute inset-0 flex items-center justify-center"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              <img
                class="rounded-lg min-h-full min-w-full flex-shrink object-cover opacity-0"
                src="/src/assets/images/construction.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 49 / 37 / span 10 / span 22;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="flex relative flex-col justify-between w-full h-full p-4 overflow-hidden rounded-xl border-[2px] bg-app-lightModeBoxes-box16 dark:bg-app-darkModeBoxes-box16 border-app-darkModeBoxes-box16 dark:border-app-lightModeBoxes-box16"
            >
              <div
                class="transition-opacity duration-500 opacity-100 h-full"
              >
                <p
                  class="text-gray-700 dark:text-gray-100 h-full"
                  data-testid="current-joke"
                >
                  I decided to create this presentation because, let's face it, my jokes arenâ€™t as good as my code... but here we are.
                </p>
              </div>
              <button
                aria-label="refresh joke"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-5fjsz0-MuiButtonBase-root-MuiIconButton-root"
                data-testid="refresh-joke-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
                  data-testid="CachedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m19 8-4 4h3c0 3.31-2.69 6-6 6-1.01 0-1.97-.25-2.8-.7l-1.46 1.46C8.97 19.54 10.43 20 12 20c4.42 0 8-3.58 8-8h3zM6 12c0-3.31 2.69-6 6-6 1.01 0 1.97.25 2.8.7l1.46-1.46C15.03 4.46 13.57 4 12 4c-4.42 0-8 3.58-8 8H1l4 4 4-4z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 30 / 37 / span 19 / span 51;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="text-blue-900 rounded-xl flex justify-center flex-wrap items-center contain-size overflow-hidden h-full border-[2px] bg-app-lightModeBoxes-box9 dark:bg-app-darkModeBoxes-box9 border-app-darkModeBoxes-box9 dark:border-app-lightModeBoxes-box9"
              data-testid="example-container"
            >
              <div
                class="absolute inset-0 flex items-center justify-center"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              <img
                class="rounded-lg min-h-full min-w-full flex-shrink object-cover opacity-0"
                src="/src/assets/images/construction.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
      style="grid-area: 21 / 43 / span 9 / span 22;"
    >
      <div
        class="flex w-full h-full"
        style="perspective: 1000px;"
      >
        <div
          class="w-full relative transition-all duration-200 ease-linear inter-var"
          style="transform-style: preserve-3d;"
        >
          <div
            class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
          >
            <div
              class="flex relative flex-col justify-between h-full p-4 rounded-xl bg-gray-200 dark:bg-gray-800"
            >
              <div
                class="mb-4"
              >
                <h2
                  class="text-xl font-bold mb-6 text-black dark:text-white"
                >
                  Hi! I'm George Gil
                </h2>
                <p
                  class="text-black dark:text-white"
                >
                  I'm a curious software engineer who also plays guitar, not always at the same time though.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
